### PR TITLE
Update Kerberos library to latest major

### PR DESCRIPTION
**Pull request by kvch**
This PR updates the Kerberos library to the latest, v8.4.2. The version used in Sarama ATM is no longer supported. All fixes and features are added to the latest major.

New import path of the lib: `github.com/jcmturner/gokrb5/v8`

### CODE REVIEW

**Comment by bai** [comment]
Thanks for contributing!

LGTM, but I'll wait for another review for safety.

**Review by bai — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -9,26 +9,18 @@ require (
...
-	github.com/jcmturner/gofork v1.0.0
-	github.com/klauspost/compress v1.11.4
-	github.com/kr/text v0.2.0 // indirect
-	github.com/pierrec/lz4 v2.6.0+incompatible
-	github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
+	github.com/frankban/quicktest v1.7.2 // indirect
+	github.com/golang/snappy v0.0.1 // indirect
+	github.com/google/go-cmp v0.4.0 // indirect
+	github.com/jcmturner/gokrb5/v8 v8.4.2
+	github.com/klauspost/compress v1.9.8
```
Comment (d1egoaz) [inline go.mod:16]: 🤔 why this version was downgraded from `v1.11.4`?

File: go.mod [hunk 2]
```diff
@@ -9,26 +9,18 @@ require (
...
-	github.com/frankban/quicktest v1.11.3 // indirect
-	github.com/golang/snappy v0.0.2 // indirect
-	github.com/hashicorp/go-uuid v1.0.2 // indirect
-	github.com/jcmturner/gofork v1.0.0
-	github.com/klauspost/compress v1.11.4
-	github.com/kr/text v0.2.0 // indirect
-	github.com/pierrec/lz4 v2.6.0+incompatible
-	github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
+	github.com/frankban/quicktest v1.7.2 // indirect
+	github.com/golang/snappy v0.0.1 // indirect
```
Comment (d1egoaz) [inline go.mod:13]: this was also downgraded

**Comment by bai** [comment]
@kvch do you think you could rebase your branch and fix the downgrade issues in go.mod/go.sum? We're about to cut a new release soon.

File: go.mod (hunk 2 above)
Comment (bai) [reply go.mod]: Fixed 🙏

File: go.mod (hunk 1 above)
Comment (bai) [reply go.mod]: Fixed 🙏

**Review by d1egoaz — APPROVED** [review]

File: go.mod (hunk 2 above)
Comment (d1egoaz) [reply go.mod]: thanks @bai

**Pull request merged**